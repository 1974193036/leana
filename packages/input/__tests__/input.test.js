import ssyyInput from "../src/input.vue"
import { mount } from "@vue/test-utils"

describe('ssyy-input', () => {
  test('input-text', () => {
    const wrapper = mount(ssyyInput)
    expect(wrapper.html()).toContain('input type="text"') // 是否包含
  })

  test('ssyy-input-password',()=>{
    const wrapper = mount(ssyyInput,{
      propsData:{
        type:'password'
      }
    })
    expect(wrapper.html()).toContain('input type="password"') // 是否包含
  })

  test('ssyy-input-password',()=>{
    const wrapper = mount(ssyyInput,{
      propsData:{
        type:'password'
      }
    })
    expect(wrapper.props('type')).toBe('password') // 是否相等prop
  })

  test('ssyy-input-snapshot', () => {
    const wrapper = mount(ssyyInput, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot() // 生成快照
  })
})