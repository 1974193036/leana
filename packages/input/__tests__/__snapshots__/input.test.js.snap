// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssyy-input ssyy-input-snapshot 1`] = `
<div>
  <input
    class="el-input__inner"
    type="text"
  />
</div>
`;
